src/groupByKey.ts

```diff
@@ -2,6 +2,13 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((groups: GroupsMap<T>, item: T) => {
+    const groupKey = item[key] as string;
+
+    groups[groupKey] = groups[groupKey] || [];
+    groups[groupKey].push(item);
+
+    return groups;
```

Comment:
The 'reduce' callback function's return type is implicitly 'GroupsMap<T>'. It's a good practice to explicitly type the return to avoid any unintended type coercion.